src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1689,6 +1729,27 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
                 }
             }
 
+            if (win32Resources != null)
+            {
+                if (!win32Resources.CanRead || !win32Resources.CanSeek)
+                {
+                    throw new ArgumentException(CodeAnalysisResources.StreamMustSupportReadAndSeek, nameof(win32Resources));
+                }
+            }
+
+            if (sourceLinkStream != null)
+            {
+                if (options?.DebugInformationFormat == DebugInformationFormat.Pdb)
+                {
+                    throw new ArgumentException(CodeAnalysisResources.SourceLinkRequiresPortablePdb, nameof(sourceLinkStream));
+                }
```

Comment:
It doesn't. In that case source link would just be ignored. I guess we should throw.